### ISSUE TITLE
Make kubectl macro testable

### ISSUE THREAD

**Issue by morningspace** [morningspace/kubemacro#4]
To ensure the quality of a kubectl macro, there needs to be ways to test it.

Typically, a kubectl macro is written in bash shell, which could be tested using [bats-core](https://github.com/bats-core/bats-core). Whenever there are needs to support kubectl macro written in other programming languages, such as Python, NodeJS, etc. they all have ways to write unit test. Also, since since kubectl macro is aimed to manipulate Kubernetes, some test harness specific to Kubernetes such as [KUTTL](https://kuttl.dev/), may also be an option.

The bottom line is that it requires some guidelines for the author of kubectl macro on how to ensure macro quality using corresponding testing techniques.
